service/gcsutils/gcstools/netnsConfig.go

```diff
@@ -155,14 +172,42 @@ func netnsConfig() error {
 			}
 		}
 	} else {
-		err := exec.Command(
-			"udhcpc",
-			"-q",
-			"-i", *ifStr,
-			"-s", "/sbin/udhcpc_config.script").Run()
+		log.Infof("Execing udhcpc with timeout...")
+		cmd := exec.Command("udhcpc", "-q", "-i", *ifStr, "-s", "/sbin/udhcpc_config.script")
+
+		done := make(chan error)
+		go func() {
+			done <- cmd.Wait()
+		}()
+
+		select {
+		case <-time.After(time.Duration(30 * time.Second)):
+			var cos string
+			co, err := cmd.CombinedOutput() // In case it has written something
+			if err != nil {
+				cos = string(co[:])
+			}
+			cmd.Process.Kill()
+			log.Infof("udhcpc timed out [%s]", cos)
+			return fmt.Errorf("udhcpc timed out. Failed to get DHCP address: %s", cos)
+		case err := <-done:
```

Comment:
Do you want to log this error at least? You immediately overwrite it with the call to .CombinedOutput

Reply:
I don't think it's worth it.